### ISSUE TITLE
Evaluation of model performance

### ISSUE THREAD

**Issue by AlfonsoAllen** [ibartomeus/Observ_models#4]
Hi guys!

I start this issue, in order to adopt a consensual protocol to evaluate model performance.

To the best of my knowledge we will deal with (i) [frequentist and Bayesian] hypothesis-driven models, (ii) machine learning (ML) models, and (iii) mechanistic models (... maybe (iv) agent-based models?). In order to estimate and predict magnitudes such as total yield, regression models are needed (at least for cases (i) and (ii)).

As far as I know, the main techniques for evaluating regression models are mainly based on the mean absolute error, median absolute error and (root) mean squared error [1]. Since we are dealing with Bayesian and frequentist models, notice that I did not mention above the determination coefficient R-squared [2], the Widely applicable Bayesian information criterion (WBIC), the Bayesian information criterion (BIC), etc.

Query 1: Do you know any alternative parameters that can be used? For instance, I think that AIC (Akaike information criterion) works fine with both (Bayesian and frequentist) approaches, but I’m not sure.

Regarding model selection and model comparisons, I suggest a classical approach for models (i) and (ii), that is

1) To split our dataset into two parts, namely: training set (70-80% of the observations) and an independent test set (30-20% of the observations).

2) To apply a k-fold cross-validation (CV) on the training set for each model and ML hyperparameter (or tuning) configuration. 
Query 2: I suggest fixing k equal to 5 or 10. What do you think?

3) To rank the models and select the best-performing. To do so, we should establish previously a selection criterion based on CV estimates of the test error for each model (mean absolute error, median absolute error and (root) or mean squared error). Since large errors are particularly undesirable, I propose using the root mean squared error instead of the mean absolute error.
Query 3: Any suggestions?

4) To use the complete training set for models fitting.
Query 4: Is this step correct?

5) To assess the predictive error of the models that we obtained previously, by making use of the independent test set and the selection criteria (see step 3). 

In the case of mechanistic models, assessing the predictive error reduces to step 5, isn’t it?

Please let me know your thoughts.

Best!

[1] An example of error validation generated by Auto-Weka.

Correlation coefficient                  0.8312
Mean absolute error                      0.4138
Root mean squared error                  0.5601
Relative absolute error                 51.8658 %
Root relative squared error             55.603  %
Total Number of Instances               93     

[2] Bayesian regression fits require a different definition of such coefficient, see Gelman (2019): https://www.tandfonline.com/doi/full/10.1080/00031305.2018.1549100
